File: benchmarks/go_comparison/logic.go
package main

import "fmt"

func main() {
	var value int64 = 1
	var iteration int64
	for iteration < 20_000_000 {
		value = value + iteration + 1
		if value > 1_000_000_000 {
			value -= 1_000_000_000
		}
		iteration++
	}
	fmt.Println(value)
}

File: benchmarks/rust_comparison/logic.rs
fn main() {
    let mut value: i64 = 1;
    let mut iteration: i64 = 0;
    while iteration < 20_000_000 {
        value = value + iteration + 1;
        if value > 1_000_000_000 {
            value -= 1_000_000_000;
        }
        iteration += 1;
    }
    println!("{value}");
}
